Typescript 5.0.X support

### Scope

Adds a new behavior

### Compatibility

- [ ] This is a breaking change

### Feature description

Typescript 5.0.2 is out

could you please update the dependency for:

`npm ERR!   peerOptional typescript@">= 4.4.x <= 4.9.x" from msw@1.1.0`

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -57,7 +57,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ts: ['4.4', '4.5', '4.6', '4.7', '4.8', '4.9']
+        ts: ['4.4', '4.5', '4.6', '4.7', '4.8', '4.9', '5.0']
     steps:
       - name: Checkout
         uses: actions/checkout@v3

--- a/package.json
+++ b/package.json
@@ -151,7 +151,7 @@
     "statuses": "^2.0.0",
     "ts-node": "^10.9.1",
     "tsup": "^5.12.8",
-    "typescript": "^5.0.0",
+    "typescript": "^5.0.2",
     "url-loader": "^4.1.1",
     "webpack": "^5.68.0",
     "webpack-dev-server": "^3.11.2",

--- a/package.json
+++ b/package.json
@@ -151,14 +151,14 @@
     "statuses": "^2.0.0",
     "ts-node": "^10.9.1",
     "tsup": "^5.12.8",
-    "typescript": "^4.9.3",
+    "typescript": "^5.0.0",
     "url-loader": "^4.1.1",
     "webpack": "^5.68.0",
     "webpack-dev-server": "^3.11.2",
     "webpack-http-server": "^0.5.0"
   },
   "peerDependencies": {
-    "typescript": ">= 4.4.x <= 4.9.x"
+    "typescript": ">= 4.4.x <= 5.0.x"
   },
   "peerDependenciesMeta": {
     "typescript": {

--- a/test/typings/tsconfig.5.0.json
+++ b/test/typings/tsconfig.5.0.json
@@ -1,0 +1,3 @@
+{
+  "extends": "./tsconfig.4.7.json"
+}

